x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/CrossClusterAccessTransportInterceptor.java

```diff
@@ -308,7 +331,11 @@ private <T extends TransportResponse> void sendWithCrossClusterAccessHeaders(
                     handler
                 );
                 try (ThreadContext.StoredContext ignored = threadContext.stashContextPreservingRequestHeaders(AuditUtil.AUDIT_REQUEST_ID)) {
-                    crossClusterAccessHeaders.writeToContext(threadContext);
+                    if (connection.getTransportVersion().supports(ADD_CROSS_CLUSTER_API_KEY_SIGNATURE)) {
+                        crossClusterAccessHeaders.writeToContext(threadContext, signer);
+                    } else {
+                        crossClusterAccessHeaders.writeToContext(threadContext, null);
```

Comment:
Passing a `null` signer is the same as "do not generate a signature", not very explicit but probably acceptable. Even if a signature is configured there is no use in providing it unless the fulfilling cluster can validate it. I considered failing here but this could actually happen in production while upgrading and we don't want to fail requests during a rolling upgrade for example.